extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzSchedulerImpl.java

```diff
@@ -195,33 +210,34 @@ public QuartzSchedulerImpl(SchedulerContext context, QuartzSupport quartzSupport
                                 }
                             }
                             CronScheduleBuilder cronScheduleBuilder = CronScheduleBuilder.cronSchedule(cron);
-                            if (perJobConfig != null) {
-                                switch (perJobConfig.misfirePolicy) {
-                                    case SMART_POLICY:
-                                        // this is the default, doing nothing
-                                        break;
-                                    case IGNORE_MISFIRE_POLICY:
-                                        cronScheduleBuilder.withMisfireHandlingInstructionIgnoreMisfires();
-                                        break;
-                                    case FIRE_NOW:
-                                        cronScheduleBuilder.withMisfireHandlingInstructionFireAndProceed();
-                                        break;
-                                    case CRON_TRIGGER_DO_NOTHING:
-                                        cronScheduleBuilder.withMisfireHandlingInstructionDoNothing();
-                                        break;
-                                    case SIMPLE_TRIGGER_RESCHEDULE_NOW_WITH_EXISTING_REPEAT_COUNT:
-                                    case SIMPLE_TRIGGER_RESCHEDULE_NOW_WITH_REMAINING_REPEAT_COUNT:
-                                    case SIMPLE_TRIGGER_RESCHEDULE_NEXT_WITH_EXISTING_COUNT:
-                                    case SIMPLE_TRIGGER_RESCHEDULE_NEXT_WITH_REMAINING_COUNT:
-                                        throw new IllegalArgumentException("Cron job " + identity
-                                                + " configured with invalid misfire policy "
-                                                + perJobConfig.misfirePolicy.dashedName() +
-                                                "\nValid options are: "
-                                                + QuartzMisfirePolicy.validCronValues().stream()
-                                                        .map(QuartzMisfirePolicy::dashedName)
-                                                        .collect(Collectors.joining(", ")));
-                                }
+                            QuartzRuntimeConfig.QuartzMisfirePolicyConfig perJobConfig = runtimeConfig.misfirePolicyPerJobs
+                                    .getOrDefault(identity, cronTriggerConfig.misfirePolicyConfig);
+                            switch (perJobConfig.misfirePolicy) {
+                                case SMART_POLICY:
+                                    // this is the default, doing nothing
```

Comment:
While this is true for the current Quartz version, nothing guarantees that it'll remain the same in the future quartz versions.
Can we ensure that we do not regress when the upstream changes by either:
- set the value here and remove this comment
- or add a test case for this?

Reply:
CronScheduleBuilder doesn't have any way of explicitly setting the misfireInstruction to MISFIRE_INSTRUCTION_SMART_POLICY

We actually do have a test already for the default smart policy for the simpleTrigger io.quarkus.quartz.test.MisfirePolicyTest#testDefaultMisfirePolicy
added a similar test for cronTrigger as well